app/stable_appconfig.go

```diff
@@ -67,8 +67,15 @@ func (app *RegenApp) registerUpgradeHandlers() {
 			"ibc":          1,
 			"genutil":      1,
 			"transfer":     1,
+			"ecocredit":    1, // we dont run InitiGenesis for ecocredit, right?
 		}
 
+		var params ecocredittypes.Params
```

Comment:
I am not sure if it makes sense.

Reply:
Setting params doesn't make much sense, we should just simply run InitGenesis on ecocredit. It makes sense to skip InitGenesis as you did on L70, and instead run it manually.

So we can replace these 4 lines with:
```go
		// force genesis to run so that ecocredit's upgrade handler will execute
		gen := ecocredittypes.DefaultGenesisState()
		gen.Params.AllowlistEnabled = true
		app.mm.Modules[ecocredittypes.ModuleName].InitGenesis(ctx, app.appCodec, app.cdc.MustMarshalJSON(gen))
```